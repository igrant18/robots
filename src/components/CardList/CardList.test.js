import '@testing-library/jest-dom';
// import React so you can use JSX (React.createElement) in your test
import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import CardList from './CardList';

describe('CardList Component Tests', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'John Snow',
            email: 'john@gmail.com'
        }
    ];
    
    it('Renders Correctly with No Props', () => {
        const tree = renderer.create(<CardList robots={mockRobots} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});