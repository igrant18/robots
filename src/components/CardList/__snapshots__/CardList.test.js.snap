// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList Component Tests Renders Correctly with No Props 1`] = `
<div>
  <div
    className="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
  >
    <img
      alt="robot"
      data-testid="id-img-element"
      src="https://robohash.org/1?size=200x200"
    />
    <div>
      <h2
        data-testid="name-h2-element"
      >
        John Snow
      </h2>
      <p
        data-testid="email-p-element"
      >
        john@gmail.com
      </p>
    </div>
  </div>
</div>
`;
